Fix prepare release process

aanm opened the pull request:
cc @michi-covalent do we need to update the steps in https://github.com/cilium/release?

aanm committed:
Revert "Update start-release.sh"

While doing the RC from `master` branch we don't want to commit all
files in a single commit since we want to have a dedicated commit for
AUTHORS update and another for the release preparation. Thus we need to
revert commit 1f1926dddc860a5bca6ccc5b34f1d8c1c2a35dd9.

Signed-off-by: André Martins <andre@cilium.io>

aanm committed:
contrib/release: add missing -C to select install/kubernetes directory

Fixes: 147c66e1ca84 ("install/kubernetes: Re-order lines in Makefile.values")
Signed-off-by: André Martins <andre@cilium.io>

aanm commented on install/kubernetes/Makefile.values:
```diff
@@ -9,7 +9,7 @@ ifeq ($(RELEASE),yes)
     export CILIUM_REPO:=quay.io/cilium/cilium
     export CLUSTERMESH_APISERVER_REPO:=quay.io/cilium/clustermesh-apiserver
-    export HUBBLE_RELAY_REPO:=quay.io/cilium/hubble-rela
+    export HUBBLE_RELAY_REPO:=quay.io/cilium/hubble-relay
```
@joestringer I noticed this after tagging v1.13.0-rc3. This is fixed in the helm chart.

joestringer commented on contrib/release/start-release.sh:
```diff
@@ -113,12 +113,12 @@ main() {
+    logecho "* Check all changes and add to a new commit"
     logecho "  * If this is a prerelease, create a revert commit"
     logecho "* Push the PR to Github for review ('submit-release.sh')"
+    logecho "* Close https://github.com/cilium/cilium/projects/$old_proj"
```
This was already done by the cilium/release script before we even run this script right?

aanm replied:
I think yes, I'll remove this line

aanm committed:
contrib/release: remove duplicated instructions

Remove manual instructions that are already performed by the automation
script.

Signed-off-by: André Martins <andre@cilium.io>

joestringer commented on contrib/release/start-release.sh:
```diff
@@ -113,12 +113,12 @@ main() {
     else
       $DIR/prep-changelog.sh "$old_version" "$version"
     fi
-    git commit -a -s -m "Prepare for release $version"
```
I'll probably just fix this issue by making it commit the authors separately during the next RC prep. Do we need to revert this?

joestringer reviewed (commented):
Commit 2, 3 makes sense but I was thinking I might just fix the issue that makes commit 1 necessary. I would like us to be moving more and more to an automated process like that rather than pushing more in the direction of requiring a human to be in the process at each step.

joestringer reviewed (approved):
We could alternatively just merge this and I'll separately include re-enabling the commit + another commit for authors etc. in a new PR after I do the next RC
